### ISSUE TITLE
Heavy slowdown when using XLFD and enabling the bar

### ISSUE THREAD

**Issue by truecolour** [conformal/spectrwm#88]
I am running spectrwm 2.6.2 on Arch Linux (Xorg 1.16.3, open source 'radeon' driver), with the default configuration. When open a xterm, spectrwm takes around a second to react. Closing windows is even slower (~1 second for it to disappear, and another ~0.5 seconds until the other windows are resized). Changing focus also seems to be affected (hard to tell).

Setting `bar_format` to a single `x` improves things; performance gets worse when I add more characters. The problem vanishes completely when I disable the bar, or set `bar_font` to a Xft font. I tried using a different XLFD, but this did not make a difference. I set my terminal (urxvt) to use a XLFD; that also had no effect.

So it looks like the way in which spectrwm uses the X11 font rendering is the culprit here.

**Comment by LordReg** [comment]
What is your bar_action?

**Comment by truecolour** [comment]
I am using the default config; judging from the comments there it is `baraction.sh`. There is no such file in my `$PATH`.

**Comment by LordReg** [comment]
I tested the community/spectrwm 2.6.2-1 package on 3 different updated Arch Linux boxes and could not reproduce this issue.  They all are running Xorg 1.16.3 and each use a different display driver: 'i915', 'nvidia' and 'radeon'.  I used the default /etc/spectrwm.conf included with the package and also tried explicitly setting an XLFD bar_font.  I'm using the radeon driver with a Radeon Mobility x1700 (RV530).

What GPU do you have? Are you using any custom xorg.conf settings for the radeon driver?  Also, have you tried the proprietary Catalyst driver? https://wiki.archlinux.org/index.php/AMD_Catalyst#Installing_from_the_unofficial_repository

**Comment by truecolour** [comment]
I did a system update this morning, and now I cannot reproduce it either. I tried various things but to no avail. Looks like the issue is gone for good.

In case this does crop up again: The graphics card is a Radeon HD5770. I am using two displays (1920x1080 and 1280x1024). The only Xorg settings I have set the primary monitor, and the relative position of the monitors. The proprietary driver had such bad 2D performance in general that I did not bother try it here.
